app/controllers/names_controller/index.rb

```diff
@@ -0,0 +1,200 @@
+    @help = :needed_descriptions_help
+    query = Name.descriptions_needed
+    show_selected_names(query, num_per_page: 100)
+  end
+
+  # Display list of names that match a string.
+  def name_search # rubocop:disable Metrics/AbcSize
+    pattern = params[:pattern].to_s
+    if pattern.match(/^\d+$/) &&
+       (name = Name.safe_find(pattern))
+      redirect_to(action: :show, id: name.id)
+    else
+      search = PatternSearch::Name.new(pattern)
+      if search.errors.any?
+        search.errors.each do |error|
+          flash_error(error.to_s)
+        end
+        render(action: :list_names)
+      else
+        @suggest_alternate_spellings = search.query.params[:pattern]
+        show_selected_names(search.query)
+      end
+    end
+  end
+
+  # Displays list of advanced search results.
+  def advanced_search
+    return if handle_advanced_search_invalid_q_param?
+
+    query = find_query(:Name)
+    show_selected_names(query)
+  rescue StandardError => e
+    flash_error(e.to_s) if e.present?
+    redirect_to(controller: :search, action: :advanced)
+  end
+
+  # Used to test pagination.
+  def test_index
+    query = find_query(:Name)
+    raise("Missing query: #{params[:q]}") unless query
+
+    if params[:test_anchor]
+      @test_pagination_args = { anchor: params[:test_anchor] }
+    end
+    show_selected_names(query, num_per_page: params[:num_per_page].to_i)
+  end
+
+  # Show selected search results as a list with 'list_names' template.
+  # cop disabled per https://github.com/MushroomObserver/mushroom-observer/pull/1060#issuecomment-1179410808
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  def show_selected_names(query, args = {})
+    store_query_in_session(query)
+    @links ||= []
+    args = {
+      action: :index,
+      letters: "names.sort_name",
+      num_per_page: (/^[a-z]/i.match?(params[:letter].to_s) ? 500 : 50)
+    }.merge(args)
+
+    # Tired of not having an easy link to list_names.
+    if query.flavor == :with_observations
+      @links << [:all_objects.t(type: :name), { action: :index }]
+    end
+
+    # Add some alternate sorting criteria.
+    args[:sorting_links] = [
+      ["name", :sort_by_name.t],
+      ["created_at", :sort_by_created_at.t],
+      [(query.flavor == :by_rss_log ? "rss_log" : "updated_at"),
+       :sort_by_updated_at.t],
+      ["num_views", :sort_by_num_views.t]
+    ]
+
+    # Add "show observations" link if this query can be coerced into an
+    # observation query.
+    @links << coerced_query_link(query, Observation)
+
+    # Add "show descriptions" link if this query can be coerced into a
+    # description query.
+    # FIXME: This link makes no sense. Needs a name_id?
+    if query.coercable?(:NameDescription)
+      @links << [:show_objects.t(type: :description),
+                 add_query_param({ controller: :name_descriptions,
+                                   action: :index },
+                                 query)]
+    end
+
```

Comment:
@JoeCohen @pellaea  — 

In this block generating the right-tab `@links` for a Names index, I don't understand what the above and below blocks are intended do here: the ones generating links that should be added to the instance variable array if `query.coercable?(:NameDescription)` or if `query.flavor == :with_descriptions`.

First of all, I can't find where these links are output in any Name index i'm able to get by interacting with the front end. But i'm probably missing some kind of unfamiliar interaction. 

Also relevant, with this refactor, is the fact that any `NameDescriptionsController#index` requires a `name_id` - there's no such thing as an index of NameDescriptions per se, if there ever was. Whatever the above link is supposed to generate, it can't do it under this refactor without a `name_id`.